cumulusci/tasks/salesforce/update_profile.py

```diff
@@ -47,168 +65,190 @@ def _init_options(self, kwargs):
         if self.options["namespaced_org"]:
             self.options["managed"] = True
 
-        # Set up namespace prefix strings
-        namespace_prefix = "{}__".format(
-            self.project_config.project__package__namespace
+        self.options["namespace_inject"] = self.options.get(
+            "namespace_inject", self.project_config.project__package__namespace
         )
+
+        # Set up namespace prefix strings
+        namespace_prefix = "{}__".format(self.options["namespace_inject"])
         self.namespace_prefixes = {
             "managed": namespace_prefix if self.options["managed"] else "",
             "namespaced_org": namespace_prefix
             if self.options["namespaced_org"]
             else "",
         }
 
-        self.profile_name = self.options.get("profile_name") or "Admin"
-
-    def _run_task(self):
-        self.retrieve_dir = None
-        self.deploy_dir = None
-        with tempfile.TemporaryDirectory() as tempdir:
-            self._create_directories(tempdir)
-            self._retrieve_unpackaged()
-            self._process_metadata()
-            self._deploy_metadata()
-
-    def _create_directories(self, tempdir):
-        self.retrieve_dir = Path(tempdir, "retrieve")
-        self.deploy_dir = Path(tempdir, "deploy")
-        self.retrieve_dir.mkdir()
-        self.deploy_dir.mkdir()
-
-    def _get_retrieve_package_xml_content(self):
-        path = self.options.get("package_xml") or os.path.join(
-            CUMULUSCI_PATH, "cumulusci", "files", "admin_profile.xml"
-        )
-        with open(path, "r") as f:
-            package_xml_content = f.read()
+        # We enable new functionality to extend the package.xml to packaged objects
+        # by default only if we meet specific requirements: the project has to require
+        # CumulusCI >= 3.8.1 (i.e., creation date or opt-in after release), and it must
+        # not be using a custom `package.xml`
+        min_cci_version = self.project_config.minimum_cumulusci_version
+        if min_cci_version and "package_xml" not in self.options:
+            parsed_version = pkg_resources.parse_version(min_cci_version)
+            default_packages_arg = parsed_version >= pkg_resources.parse_version(
+                "3.8.1"
```

Comment:
@davisagli  Would the pre-trail release go out as 3.8.1 or as 3.9.0? (I suppose this will work either way, but for clarity's sake).

Reply:
Depends on whether the release includes new features. Probably 3.9.0. You can leave this is, put a note in the PR description to remind me, and I'll adjust when I prepare the release notes.